src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/service/SpeakerFlowSegmentRequestBuilder.java

```diff
@@ -683,6 +690,52 @@ private Optional<MirrorConfig> makeMirrorConfig(@NonNull FlowPath flowPath, @Non
         return Optional.empty();
     }
 
+    private Set<MirrorConfigData> getMirrorConfigData(
+            FlowMirrorPoints flowMirrorPoints, FlowEncapsulationType flowEncapsulationType) {
+        Map<PathId, EncapsulationResources> encapsulationMap = resourcesManager.getMirrorEncapsulationMap(
+                Lists.newArrayList(flowMirrorPoints), flowEncapsulationType);
+        Set<MirrorConfigData> result = new HashSet<>();
+        for (FlowMirror flowMirror : flowMirrorPoints.getFlowMirrors()) {
+            for (FlowMirrorPath mirrorPath : flowMirror.getMirrorPaths()) {
+                if (mirrorPath.isForward() && !mirrorPath.isDummy()) {
+                    if (mirrorPath.isSingleSwitchPath()) {
+                        result.add(new MirrorConfigData(
+                                flowMirror.getEgressPort(), flowMirror.getEgressOuterVlan(), null));    
+                    } else {
+                        buildMultiSwitchMirrorConfig(mirrorPath, flowEncapsulationType, encapsulationMap)
+                                .ifPresent(result::add);
+                    }
+                }
+            }
+        }
+        return result;
+    }
+    
+    private static Optional<MirrorConfigData> buildMultiSwitchMirrorConfig(
+            FlowMirrorPath mirrorPath, FlowEncapsulationType flowEncapsulationType,
+            Map<PathId, EncapsulationResources> mirrorEncapsulationMap) {
+        if (mirrorPath.getSegments() == null || mirrorPath.getSegments().isEmpty()) {
+            log.error(format("Multi switch flow mirror path %s has no segments",
```

Comment:
A minor thing, but still: "multi" is not a word and cannot be alone; it is either multiswitch as a noun, meaning multiswitch is the name of a type of a device, or it is multi-switch _something_, meaning multi-switch here is a compound adjective. I believe it is the second case here: it is a characteristic of a flow, right?

Reply:
fixed